jupyter_server_proxy/config.py

```diff
@@ -208,26 +209,35 @@ class ServerProxy(Configurable):
 
           rewrite_response
             An optional function to rewrite the response for the given service.
-            Input arguments are ``host`` which is ``"localhost"``, the service
-            port ``port``, the ``path`` from the requested URL, and
-            ``response`` is a `tornado.httpclient.HTTPResponse object
-            <https://www.tornadoweb.org/en/stable/httpclient.html#response-objects>`.
-            Output is bytes.
-            Defaults to ``lambda host, port, path, response: response.body``.
+            Input is a RewritableResponse object. The function should modify one or
```

Comment:
Input should also be given the request object if it asks for it.

Reply:
@yuvipanda, does my latest commit properly address how to access the request object?

Reply:
Can we pass them as params when asked for instead? I am generally a big fan of not creating custom class hierarchies unless required :D

Reply:
Hmm, I thought you and minrk just gave a thumbs-up to my custom `RewritableResponse` class. It is supposed to be a container for all the necessary arguments. If you don't like it then we have to start from scratch, right? Or maybe I'm misunderstanding what you're saying.

(I have a very strong distaste myself for excessive OOP, but here it seems to me like it serves a good purpose.)

Reply:
@maresb so my understanding is that there are *two things* - a `Response` that is coming back from our upstream server that we are rewriting (and hence the `RewritableResponse`, but also a readonly *request* that came from the client that triggered this response eventually. Right now, I think that request is inside this response object, even though from a pure HTTP perspective they aren't related. So in my mind, the function signature would be something like `def rewrite(orig_request: tornado.web.HTTPServerRequest, response: RewriteableResponse)` - hope that makes sense? So this shouldn't be starting from scratch as much as making `RewriteableResponse` a pure analogue of a hypothetical `Response` object. The original_request is readonly so shouldn't affect chaining.